tools/testing/e2e/test_utils.h

```diff
@@ -0,0 +1,154 @@
+#ifndef IREE_TESTING_COMMON_UTILS_H_
+#define IREE_TESTING_COMMON_UTILS_H_
```

Comment:
match the define to the file name, so `IREE_TOOLS_TESTING_E2E_TEST_UTILS`

Otherwise if someone adds a `testing/common_utils.h` they might accidently use the same header guard and only one file would be defined x_x